exporter/exporterhelper/internal/queuebatch/queue_batch.go

```diff
@@ -42,32 +43,35 @@ func NewQueueBatch(
 		b = newDefaultBatcher(*cfg.Batch, next, cfg.NumConsumers)
 	}
 
-	sizer, ok := qSet.Sizers[request.SizerTypeRequests]
-	if !ok {
-		return nil, errors.New("queue_batch: unsupported sizer")
-	}
-
 	var q Queue[request.Request]
 	switch {
 	case cfg.WaitForResult:
 		q = newDisabledQueue(b.Consume)
-	case cfg.StorageID != nil:
-		q = newAsyncQueue(newPersistentQueue[request.Request](persistentQueueSettings[request.Request]{
-			sizer:     sizer,
-			capacity:  int64(cfg.QueueSize),
-			blocking:  cfg.BlockOnOverflow,
-			signal:    qSet.Signal,
-			storageID: *cfg.StorageID,
-			encoding:  qSet.Encoding,
-			id:        qSet.ID,
-			telemetry: qSet.Telemetry,
-		}), cfg.NumConsumers, b.Consume)
 	default:
-		q = newAsyncQueue(newMemoryQueue[request.Request](memoryQueueSettings[request.Request]{
-			sizer:    sizer,
-			capacity: int64(cfg.QueueSize),
-			blocking: cfg.BlockOnOverflow,
-		}), cfg.NumConsumers, b.Consume)
+		sizer, ok := qSet.Sizers[cfg.Sizer]
+		if !ok {
+			return nil, fmt.Errorf("queue_batch: unsupported sizer %q", cfg.Sizer)
+		}
+
+		switch cfg.StorageID != nil {
+		case true:
```

Comment:
not a blocker, but since there's only one case, do we need a switch here? (same with the switch at 47)